Sources/Player/PlaybackEngine/Internal/AVQueuePlayer/AVQueuePlayerWrapper.swift

```diff
@@ -594,19 +590,17 @@ private extension AVQueuePlayerWrapper {
 			// AVPlayer has moved to the next item in its queue
 			if let currentPlayerItem = self.player.currentItem {
 				self.delegates.completed(asset: asset)
-
 				if self.player.timeControlStatus == .playing {
 					self.playing(playerItem: currentPlayerItem)
 				}
-
 			} else {
-				// AVPlayer had no other item in the queue (we are still downloading the next one)
+				// AVPlayer had no other item in the queue (we are still downloading the next one?)
```

Comment:
Is this comment a question now?

Reply:
he, unclear what I mean there lol.

There is an `if/else` right after. So it depends on the `if` condition.

Would be better like this I guess
```
// AVPlayer had no other item in the queue (We might still be downloading the next one)`
```

Reply:
The reason for this is that if an `AVPlayerItem` is in the player queue, it will not be cached in advance. I'm not sure why, but these are both internal Apple APIs...

So what I do internally when `setNext` is called, is to start caching it when the current item has fully downloaded), but not enqueue it right away.